examples/analysis/lib/strict_modes.dart

```diff
@@ -0,0 +1,32 @@
+// ignore_for_file: dead_code, prefer_typing_uninitialized_variables
+
+import 'package:examples_util/ellipsis.dart';
+
+String strictCasts() {
+  // #docregion strict-casts
+  dynamic o = ellipsis<String>();
```

Comment:
IANATW, but my 2c:

This example is perfectly correct, but if we want to highlight a more common or nefarious case, then parameters and JSON are a great example:

```dart
// a function in one file
void foo(List<String> p) { ... }

// another function far away, different file
void bar(String jsonText) {
  foo(jsonDecode(jsonText));
}
```

This example highlights 1) how code not on your screen can declare the type to which you are casting, and 2) how the code looks fine at compile-time, but could 100% crash at runtime, because of the implicit cast.

[1] I am not a technical writer :)